bazel/grpc_build_system.bzl

```diff
@@ -28,6 +28,12 @@ load("//bazel:cc_grpc_library.bzl", "cc_grpc_library")
 # The set of pollers to test against if a test exercises polling
 POLLERS = ["epollex", "epoll1", "poll", "poll-cv"]
 
+def is_msvc():
```

Comment:
I'm wondering if the is_msvc function is actually useful - it seems that most place where it's being used you won't be able to build on windows regardless of whether MSVC (or e.g clang) is being used. So maybe for simplicity, we could just use `is_windows`, `supports_windows`, etc. - that's probably more understandable to readers and perhaps even more correct.

Reply:
there's also Cygwin gcc, which contains most headers of POSIX, also the checks inside the function checks for MSVC only.

Reply:
I will leave up to you, but to me it feels that trying to suport cygwin gcc is asking for trouble - we don't really support it, so not worth the work trying to enable it here IMHO.
this doc doesn't mention mingw or cygwin at all:
https://github.com/grpc/grpc/blob/master/BUILDING.md#grpc-c---building-from-source

Reply:
my point is not to support cygwin gcc, but limiting the Windows compilation to MSVC at the first stage.

Reply:
I really don't see why "is_windows" won't do at least as good of a job in all occurrences of is_msvc in this PR (e.g. for the tests that "uses_polling" - there's clearly no chance of getting pollers working on windows regardless of msvc being used or not - so is_windows seems a bit more correct), but I'm fine with the PR as is, we can followup in the future.

Reply:
this function is not only called in this file, but in other files as well (end2end/generate_tests.bzl), potentially when we start support WIndows clang, there might be a difference in targets compiled.